docs/cli/overview.md

```diff
@@ -0,0 +1,43 @@
+# CLI Architecture Overview
+
+## Entrypoint
+
+[cmd/kbc/main.go](../../cmd/kbc/main.go)
+
+## CLI Commands
+
+Use [Cobra framework](https://cobra.dev/).
+
+CLI commands are defined in [`internal/pkg/service/cli/cmd/`](../../internal/pkg/service/cli/cmd):
+
+- [`ci`](../../internal/pkg/service/cli/cmd/ci) - Commands to generate CI workflows for the local project.
+- [`dbt`](../../internal/pkg/service/cli/cmd/dbt) - Commands to generate definitions for dbt.
+- [`local`](../../internal/pkg/service/cli/cmd/local) - Commands to work locally inside the project folder (i.e. create a config or apply a template).
+- [`remote`](../../internal/pkg/service/cli/cmd/remote) - Commands to work remotely in the Keboola project (i.e. create a branch, create a workspace).
+- [`sync`](../../internal/pkg/service/cli/cmd/sync) - Commands to synchronize state between the local folder and its Keboola project (i.e. pull changes from the project, push changes to the project).
+- [`template`](../../internal/pkg/service/cli/cmd/template) - Commands to manage templates in the project or manage the templates repository.
+
+## Dialogs
+
+Most CLI commands offer an interactive dialog to fill in the needed information. E.g. `local create` command asks what to create 
+by calling [`d.Dialogs().AskWhatCreateLocal()`](https://github.com/keboola/keboola-as-code/blob/6157a3d111c04adc58e87485e065b386046fffaa/internal/pkg/service/cli/cmd/local/create.go#L28-L28).
+
+The dialogs are defined in the [`internal/pkg/service/cli/dialog`](../../internal/pkg/service/cli/dialog) package.
+
+They use [`internal/pkg/service/cli/prompt`](../../internal/pkg/service/cli/prompt) package that defines `Prompt` interface for the following types of user inputs:
+
+- `Confirm` - Expects `Yes` or `No` answer.
+- `Question` - Expects a typed answer.
+- `Select` - Lists options to choose one from.
+- `SelectIndex` - Lists options indexed by a number. Returns index for the selected option.
+- `MultiSelect` - Lists options and allows choosing any number of them.
+- `MultiSelectIndex` - Lists options indexed by a number and allows choosing any number of them. Returns an array of indices for the selected options.
+- `Multiline` - Allows a multi-line answer.
+
+## Operations
+
+After collecting user inputs, a CLI command typically calls an operation or a set of operations. (The operation 
+corresponds to the [`command design pattern`](https://refactoring.guru/design-patterns/command) but we decided to rename
+the commands to operations not to confuse them with the CLI commands.)
+
+The operations are defined in [`pkg/lib/operation`](../../pkg/lib/operation) folder. 
```

Comment:
To mame podrobnejsie zdokumentovane tu, mozno by to mohol byt podnadpis `### Dependency Injection`, alebo 

https://github.com/keboola/keboola-as-code/blob/7b1606b4236d077be0c912f52814ce7e6acec69f/internal/pkg/service/common/dependencies/dependencies.go#L1-L5

Reply:
Pak by mohlo stačit napsat že podrobnější info je v tom souboru? https://github.com/keboola/keboola-as-code/blob/a8732128d3f7bd677de8dfd3fcfb74cc633505d0/docs/cli/overview.md#operations

Reply:
Kludne. Len to proste nejak nalinkovat.